.github/workflows/staging-release.yaml

```diff
@@ -1,121 +1,125 @@
-        rpm --import $KEYFILE
-        rm -f $KEYFILE
+          OUTPUT=$(ssh-keyscan -H ${{ secrets.FLUENTBITIO_HOST }})
+          echo ::set-output name=OUTPUT::$OUTPUT
+
+    - name: Install SSH Key
+      uses: shimataro/ssh-key-action@v2
+      with:
+        key: ${{ secrets.FLUENTBITIO_SSHKEY }}
+        known_hosts: ${{ steps.known_hosts.outputs.OUTPUT }}
 
     - name: Sync packages from staging on S3
-      # Download the current release bucket
-      # Add everything from staging
-      # Sign and set up metadata for it all
-      # Upload to release bucket
       run: |
-        rm -rf ./releases/
-        mkdir -p ./releases/
-        if [ -n "${AWS_S3_ENDPOINT}" ]; then
-          ENDPOINT="--endpoint-url ${AWS_S3_ENDPOINT}"
-        fi
-        aws s3 sync "s3://${{ secrets.AWS_S3_BUCKET_RELEASE }}" ./releases/ --no-progress ${ENDPOINT}
-        aws s3 sync "s3://${{ secrets.AWS_S3_BUCKET_STAGING }}" ./releases/ --no-progress ${ENDPOINT}
-        VERSION=$(cat ./releases/latest-version.txt)
-
-        sudo apt-get install debsigs createrepo
-        ./update-repos.sh "$VERSION" ./releases/
-        aws s3 sync ./releases/ "s3://${{ secrets.AWS_S3_BUCKET_RELEASE }}" --follow-symlinks --no-progress ${ENDPOINT}
+        mkdir -p packaging/releases
+        aws s3 sync "s3://$BUCKET" packaging/releases/ --no-progress
       env:
-        GPG_KEY: ${{ steps.import_gpg.outputs.name }}
-        AWS_REGION: "us-east-1"
         AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
         AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-        # To use with Minio locally
-        # AWS_S3_ENDPOINT: http://localhost:9000
+        # Replace with staging bucket if required later
+        BUCKET: ${{ secrets.AWS_S3_BUCKET_RELEASE }}
       shell: bash
-      working-directory: packaging
 
-  staging-release-images:
-    runs-on: ubuntu-latest
-    environment: release
-    # Use skopeo to do all the work - run the container to keep it easy to use.
-    steps:
-      - name: Sync container images from staging to release
-        run: |
-          docker run --rm quay.io/skopeo/stable:latest \
-            sync --all \
-            --src-creds ${{ github.actor }}:${{ secrets.GITHUB_TOKEN }} \
-            --dest-creds ${{ secrets.DOCKERHUB_USERNAME }}:${{ secrets.DOCKERHUB_TOKEN }} \
-            --src docker --dest docker \
-            ${{ env.STAGING_REGISTRY }}/${{ env.STAGING_IMAGE_NAME }} \
-            ${{ env.RELEASE_REGISTRY }}//${{ env.RELEASE_IMAGE_NAME }}
-        env:
-          STAGING_REGISTRY: ghcr.io
-          STAGING_IMAGE_NAME: ${{ github.repository }}
-
-          # docker.io/fluent/fluent-bit
-          RELEASE_REGISTRY: docker.io
-          RELEASE_IMAGE_NAME: ${{ secrets.DOCKERHUB_ORGANIZATION }}
+    - name: Upload to build server
+      run: |
+        ssh $USERNAME@$HOST mkdir -p /home/$USERNAME/apt
+        rsync -avz packaging/releases/ $USERNAME@$HOST:/home/$USERNAME/apt
```

Comment:
We could also sync to a per-version directory, etc.?

Reply:
shouldn't rsync just sync the delta between what is on the builder vs the server?

Reply:
Yup.
My main question is the pull from AWS first of the whole release bucket which will keep growing.